### ISSUE TITLE
Gradients and World Coordinates

### ISSUE THREAD

**Issue by mbatesole** [nmwsharp/diffusion-net#21]
Hello, Firstly thank you so much for all of your work and releasing it in a way that even lay people like myself can make an attempt at understanding it. I'm a big fan. 

I've had no formal training in this field, so if this question comes across as just plain silly, well-- I apologize ahead of time. I know in your other work, intrinsic data is vital to "robustify-ing" the results of processes run on meshes. From what I think I understand in diffusion-net, the tangent planes and thus the gradients, are anchored in global XYZ coordinates-- which in turn is why you add small permutations to the training set. Have you considered orienting the tangent plane to the directions of principal curvature rather than the world space? Wouldn't this keep the process intrinsic to the mesh, making diffusion-net more robust against different mesh orientations? 

But maybe I'm way off base... I'm still trying to fully grasp the workings of diffusion-net. At any rate, thank you again.
